docs/Testing.md

```diff
@@ -1,13 +1,140 @@
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing-example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap pino-pretty --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+app.js
+
+```js 
+const Fastify = require('fastify')
+
+function build(opts={}) {
+  const fastify = Fastify(opts)
+  fastify.get('/', async function (request, reply) {
+    return { hello: 'world' }
+  })
+
+  return fastify
+}
+
+module.exports = build
+```
+
+server.js
+
+```js
+const server = require('./app')({
+    logger: {
+      level: 'info',
+      prettyPrint: true
+    }
+  })
+
+server.listen(3000, (err, address) => {
+  if (err) {
+    console.log(err)
+    process.exit(1)
+  }
+  server.log.info(`server listening on ${address}`)
+})
+```
+
+### Benefits of using fastify.inject()
+
 Fastify comes with built-in support for fake http injection thanks to [`light-my-request`](https://github.com/fastify/light-my-request).
 
-To inject a fake http request, use the `inject` method:
+Before introducing any tests, we'll use the `.inject` method to make a fake request to our route:
+
+app.test.js
+
+```js
+const build = require('./app')
+
+const test = async () => {
+  const fastify = build()
+  await fastify.ready()
+  
+  try {
+  const response = await fastify.inject(
+    {
+      method: 'GET',
+      url: '/'
+    }
+  )
+  console.log('status code: ', response.statusCode)
+  console.log('body: ', response.body)
+} catch (err) {
+  // handle error
+}
+
+  
+}
+test()
+```
+
+First, our code will run inside an asynchronous function, giving us access to async/await. 
+
+The `.ready` method insures all registered plugins have booted up and our application is ready to test. Lastly we call the `.inject` method passing the method we want to use and our "/" route. Using await we can store the response without a callback.
+
+
+
+Run the test file in your terminal `node app.test.js`
+
+```sh
+status code:  200
+body:  {"hello":"world"}
+```
+
+
+
+### Testing with http injection
+
+Now we can replace our console.log calls with actual tests!
+
+In your package.json change the "test" script to:
+
+`"test": "tap --reporter=list --watch"`
+
+app.test.js
+
+```js
+const { test } = require('tap')
+const build = require('./app')
+
+test('requests the "/" route', async t => {
+  const fastify = build()
+  await fastify.ready()
+
+  try {
+    const response = await fastify.inject({
+      method: 'GET',
+      url: '/'
+    })
+    t.strictEqual(response.statusCode, 200, 'returns a status code of 200')
+    t.end()
+  } catch (err) {
+    t.fail('it must not throw')
+  }
```

Comment:
remove the try-catch, it will make debugging any error worse.

Reply:
I removed it. Not sure if you wanted something else or just remove it.